test/e2e/curated_packages_test.go

```diff
@@ -4,37 +4,174 @@
-)
+var EksaPackageControllerHelmValues = []string{"sourceRegistry=public.ecr.aws/l0g8r8j6"}
+
+type resourcePredicate func(string, error) bool
+
+func NoErrorPredicate(_ string, err error) bool {
+	return err == nil
+}
+
+// TODO turn them into generics using comparable once 1.18 is allowed
+func StringMatchPredicate(s string) resourcePredicate {
+	return func(in string, err error) bool {
+		return err == nil && strings.Compare(s, in) == 0
+	}
+}
+
+func IntEqualPredicate(i int) resourcePredicate {
+	return func(in string, err error) bool {
+		inInt, err := strconv.Atoi(in)
+		return err == nil && inInt == i
+	}
+}
+
+func WaitForResource(
+	test *framework.ClusterE2ETest,
+	ctx context.Context,
+	resource string,
+	namespace string,
+	jsonpath string,
+	timeout time.Duration,
+	predicates ...resourcePredicate,
+) error {
+	start := time.Now()
+	end := start.Add(timeout)
+	if !strings.HasPrefix(jsonpath, "jsonpath") {
+		jsonpath = fmt.Sprintf("jsonpath='%s'", jsonpath)
+	}
+	command := []string{}
+	for now := start; now.Before(end); {
+		out, err := test.KubectlClient.Execute(ctx, "get", "-n", namespace,
+			"--kubeconfig="+test.Cluster().KubeconfigFile,
+			"-o", jsonpath, resource)
+		outStr := out.String()
+		trimmed := strings.Trim(outStr, "'")
+		allPredicates := true
+		for _, f := range predicates {
+			allPredicates = f(trimmed, err) && allPredicates
+		}
+		if allPredicates {
+			return nil
+		}
+		time.Sleep(2 * time.Second)
+		now = time.Now()
+	}
+	return fmt.Errorf(
+		"Timed out waiting for resource: %s [namespace: %s, jsonpath: %s, timeout: %s]",
+		command,
+		namespace,
+		jsonpath,
+		timeout,
+	)
+}
+
+func WaitForDaemonset(
+	test *framework.ClusterE2ETest,
+	ctx context.Context,
+	daemonsetName string,
+	namespace string,
+	numberOfNodes int,
+	timeout time.Duration,
+) error {
+	return WaitForResource(
+		test,
+		ctx,
+		"daemonset/"+daemonsetName,
+		namespace,
+		"{.status.numberAvailable}",
+		timeout,
+		NoErrorPredicate,
+		IntEqualPredicate(numberOfNodes),
+	)
+}
+
+// Hackish way to get the latest bundle. This assumes no bundle is created outside of the normal PBC bundle fetch timer.
+// This should be modified to get the bundle from the previous build step and use that only.
+func WaitForLatestBundleToBeAvailable(
+	test *framework.ClusterE2ETest,
+	ctx context.Context,
+	timeout time.Duration,
+) error {
+	return WaitForResource(test, ctx, "packagebundle", "eksa-packages", "{.items[0]}", timeout, NoErrorPredicate)
+}
+
+func WaitForPackageToBeInstalled(
+	test *framework.ClusterE2ETest,
+	ctx context.Context,
+	packageName string,
+	timeout time.Duration,
+) error {
+	//--for=jsonpath isn't supported in v1.22. Update once it's supported
+	//_, err = test.KubectlClient.Execute(
+	//    ctx, "wait", "--timeout", "1m",
+	//    "--for", "jsonpath='{.status.state}'=installed",
+	//    "package", packagePrefix, "--kubeconfig", kubeconfig,
+	//    "-n", "eksa-packages",
+	//)
```

Comment:
Is the PR in draft?

Reply:
It can be removed but I'd like to keep it as we'll use that implementation once the cli image includes the kubernetes v1.23 client.